src/vs/platform/update/electron-main/abstractUpdateService.ts

```diff
@@ -52,14 +52,19 @@ export abstract class AbstractUpdateService implements IUpdateService {
 		@IRequestService protected requestService: IRequestService,
 		@ILogService protected logService: ILogService,
 		@IProductService protected readonly productService: IProductService
-	) { }
+	) {
+		Promise.race([
```

Comment:
This is maybe a misleading name, but the event phase will trigger even when no window opens.

Btw I do not think you can open an application on macOS without a window opening as well, at least not with VS Code.

Reply:
Strange... I though you could... I've been away from macOS too long. 😄 What happens if you restart and update, from a no-window state? Wouldn't you also get another no-window state?

Reply:
Yeah good point, that is possible. Though again, we start the phase here:

https://github.com/microsoft/vscode/blob/815295b041a4a1d59dbe7c33a17168fef36e8138/src/vs/code/electron-main/app.ts#L560-L561

Which is code that will always run, irrespective of windows opening or not. Otherwise we would already today see an issue because your code literally runs a few lines below that.